liquidctl/driver/kraken_three_x.py

```diff
@@ -0,0 +1,327 @@
+            currently not possible to address devices via led id e.g. "led1"
+            accessory_id: ? (LED 1 - ?) & 0x10 (LED 2 - ring) & 0x11 (LED 3 - logo)
+            """
+            num_light_channels = msg[14]  # the 15th byte (index 14) is # of light channels
+            accessories_per_channel = 6  # each lighting channel supports up to 6 accessories
+            light_accessory_index = 15  # offset in msg of info about first light accessory
+            for light_channel in range(num_light_channels):
+                for accessory_num in range(accessories_per_channel):
+                    accessory_id = msg[light_accessory_index]
+                    light_accessory_index += 1
+                    if accessory_id != 0:
+                        status.append(('LED {} accessory {}'.format(light_channel + 1, accessory_num + 1),
+                                       _ACCESSORY_NAMES.get(accessory_id, 'Unknown'), ''))
+
+        self._read_until({b'\x11\x01': parse_firm_info, b'\x21\x03': parse_led_info})
+        self.device.release()
+        return sorted(status)
+
+    def get_status(self, **kwargs):
+        """Get a status report.
+
+        Returns a list of `(property, value, unit)` tuples.
+        """
+        msg = self._read()
+        return [
+            ('Liquid temperature', msg[15] + msg[14] / 10, '°C'),
+            ('Pump speed', msg[18] << 8 | msg[17], 'rpm'),
+            ('Pump duty', msg[19], '%'),
+        ]
+
+    def set_color(self, channel, mode, colors, speed='normal', **kwargs):
+        """Set the color mode for a specific channel."""
+        cid = _COLOR_CHANNELS[channel]
+        _, _, _, mincolors, maxcolors = _COLOR_MODES[mode]
+        colors = [[g, r, b] for [r, g, b] in colors]
+        if len(colors) < mincolors:
+            raise ValueError('Not enough colors for mode={}, at least {} required'.format(mode, mincolors))
+        elif maxcolors == 0:
+            if colors:
+                LOGGER.warning('too many colors for mode=%s, none needed', mode)
+            colors = [[0, 0, 0]]  # discard the input but ensure at least one step
+        elif len(colors) > maxcolors:
+            LOGGER.warning('too many colors for mode=%s, dropping to %i', mode, maxcolors)
+            colors = colors[:maxcolors]
+        sval = _ANIMATION_SPEEDS[speed]
+        self._write_colors(cid, mode, colors, sval)
+        self.device.release()
+
+    def set_speed_profile(self, channel, profile, **kwargs):
+        """Set channel to use a speed profile."""
+        if channel != 'pump':
+            assert False, 'kraken X3 devices only support changing pump speeds'
+        header = [0x72, 0x01, 0x00, 0x00]
+        norm = normalize_profile(profile, _CRITICAL_TEMPERATURE)
+        interp = [(interpolate_profile(norm, t)) for t in range(20, 60)]
+        LOGGER.debug('setting pump curve: %s', [(num+20, duty) for (num, duty) in enumerate(interp)])
+        raise NotImplementedError()  # precaution
```

Comment:
Since I am not 100% sure this works as expected, I added this error temporarily.

With the sample from the kraken 3rd gen documentation it generates this curve: 

`set pump speed 20 30  30 50  34 80  40 90  50 100`

`setting pump curve: [(20, 30), (21, 32), (22, 34), (23, 36), (24, 38), (25, 40), (26, 42), (27, 44), (28, 46), (29, 48), (30, 50), (31, 58), (32, 65), (33, 72), (34, 80), (35, 82), (36, 83), (37, 85), (38, 87), (39, 88), (40, 90), (41, 91), (42, 92), (43, 93), (44, 94), (45, 95), (46, 96), (47, 97), (48, 98), (49, 99), (50, 100), (51, 100), (52, 100), (53, 100), (54, 100), (55, 100), (56, 100), (57, 100), (58, 100), (59, 100)]`

First value is the temperature and second the duty value. Does this look ok?

The resulting write operation would be and seems valid as well:
`72 01 00 00 1e 20 22 24 26 28 2a 2c 2e 30 32 3a 41 48 50 52 53 55 57 58 5a 5b 5c 5d 5e 5f 60 61 62 63 64 64 64 64 64 64 64 64 64 64 (and 20 padding bytes)`

Reply:
It looks ok to me.